JS/main.js

```diff
@@ -28,19 +28,22 @@ function filenameToIdentifier(filename) {
 
 	// Array(...filename parts, {OP,ED}{0,1,2,...}[{a,b,c,...}], [N]C{BD,DVD,PC,...})
 	var parts = filename.split("-");
-
-	// [N]C{BD,DVD,PC,...}
-	parts.pop();
-
-	// {OP,ED}{0,1,2,...}[{a,b,c,...}]
-	var subident = parts.pop();
-	// {OP,ED}{1,2,...}[{a,b,c,...}]
-	subident = subident.replace(/(\D+)0*(.+)/, "$1$2");
-	// {Opening,Ending}{1,2,...}[{a,b,c,...}]
-	subident = subident.replace("OP", "Opening");
-	subident = subident.replace("ED", "Ending");
-
-	return subident + "-" + parts.join("-");
+        // [N]C{BD,DVD,PC,...}
+        parts.pop();
+
+        // {OP,ED}{0,1,2,...}[{a,b,c,...}]
+        var subident = parts.pop();
+        if (typeof subident != "undefined" ) {
```

Comment:
One problem here is that there's no guarantee that this will be undefined. In fact, it *won't* be undefined if the filename had more than one dash in it. I don't know how to fix this, since the previous system only required that file names not have a period in them.